moon/controller.py

```diff
@@ -479,6 +483,48 @@ def on_scan(self, data):
         before_robot.sort();
         self.scan_distance_to_obstacle = median(before_robot[:self.scan_min_window])
 
+        # unless we are already inside unsafe area, augment nearest obstacles with virtual fence values
+        if self.unsafe_location is not None and not self.unsafe_location.contains(Point(self.xyz[0], self.xyz[1])):
+            # calculate position in front of and to the left and right of the robot in the direction of its heading
+            v = np.asmatrix(np.asarray([self.xyz[0], self.xyz[1], 1]))
+            c, s = np.cos(self.steering_angle + self.yaw), np.sin(self.steering_angle + self.yaw)
+            R = np.asmatrix(np.array(((c, -s, 0), (s, c, 0), (0, 0, 1))))
+
+            T = np.asmatrix(np.array(((1, 0, +6),(0, 1, 0),(0, 0, 1))))
```

Comment:
I guess that this is turned location at 6 meters? I would maybe add it as contstant not to mix with identity matrix?

Reply:
This is 6m ahead in the direction of steering, then we also look 6m ahead 30 lidar samples (ie 0.78rad) to the left and to the right (see above).